src/openforms/appointments/base.py

```diff
@@ -220,3 +224,16 @@ def get_appointment_details_html(self, identifier: str) -> str:
         return render_to_string(
             "appointments/appointment_details.html", {"appointment": details}
         )
+
+    def get_appointment_links_html(self, submission: Submission) -> str:
+
+        token = submission_appointment_token_generator.make_token(submission)
+        cancel_url = reverse(
+            "api:appointments-verify-cancel-appointment-link",
+            kwargs={"token": token, "submission_id": submission.id},
+        )
```

Comment:
_please_ base64 encode the submission ID to not plainly expose database IDs. I'd even say to just base64 the submission UUID instead of using the database PK. See also the django password reset URLs: https://github.com/django/django/blob/3921b1c6d24c9d5a60e5f5f83c9a394104089c21/django/contrib/auth/views.py#L270